BetterPasswordPlugin.inc.php

```diff
@@ -160,24 +160,52 @@ function checkPassword($hookName, $args) {
 		if ($this->getSetting(CONTEXT_SITE, 'betterPasswordCheckBlacklist')) {
 			$badPassword = false;
 			$lowerPassword = strtolower($password);
-			foreach ($this->getBlacklists() as $filename) {
-				/* In case of out-of-memory error, break glass
-				$passwordfile = fopen($filename, "r");
-				while (!feof($passwordfile)) {
-					$disallowed = fgets($passwordfile);
-					if ($lowerPassword === $disallowed) {
-						$badPassword = true;
-						break;
-					}
-				}
-				fclose($passwordfile);
-				 */
-				$passwordfile = file_get_contents($filename);
-				if (strpos($passwordfile, $lowerPassword) !== false) {
+                        $sha_pass = sha1($lowerPassword);
+                        $password_hash = substr(sha1($lowerPassword),0,2);
+                        $blacklistFileSetting = $this->getSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles');
+                        if (is_null($blacklistFileSetting)) {
+                            $blacklistFileSettingHash = array();
+                            foreach ($this->getBlacklists() as $filename) {
+                                $blacklistFileSettingHash[$filename] = sha1_file($filename);
+                            }
+                            $this->updateSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles', $blacklistFileSettingHash);
+                        }
+                        else {
+                            $currBlacklist = array();
+                            foreach ($this->getBlacklists() as $filename) {
+                                $currBlacklist[$filename] = sha1_file($filename);
+                            }
+                            $prevBlacklist = $this->getSetting(CONTEXT_SITE, 'betterPasswordBlacklistFiles');
```

Comment:
Because these are associative arrays, PHP makes this easy on you:
https://www.php.net/manual/en/language.operators.array.php
`$prevBlackList != $currBlackList`

But what if your new helper function to manage the setting also returned true or false to indicate if it had to do anything?